### PR TITLE
XGO-Core

### PR DESCRIPTION
**Pull request by Marton6**
This PR should be merged when we decide to release the (beta) .NET client.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,83 +0,0 @@
...
-golang.org/x/tools v0.0.0-20200117012304-6edc0a871e69/go.mod h1:TB2adYChydJhpapKDTa4BR/hXlZSLoq2Wpct/0txZ28=
-golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
-golang.org/x/xerrors v0.0.0-20191011141410-1b5146add898/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
-golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
-google.golang.org/protobuf v0.0.0-20200109180630-ec00e32a8dfd/go.mod h1:DFci5gLYBciE7Vtevhsrf46CRTquxDuWsQurQQe4oz8=
-google.golang.org/protobuf v0.0.0-20200221191635-4d8936d0db64/go.mod h1:kwYJMbMJ01Woi6D6+Kah6886xMZcty6N08ah7+eCXa0=
-google.golang.org/protobuf v0.0.0-20200228230310-ab0ca4ff8a60/go.mod h1:cfTl7dwQJ+fmap5saPgwCLgHXTUD7jkjRqWcaiX5VyM=
-google.golang.org/protobuf v1.20.1-0.20200309200217-e05f789c0967/go.mod h1:A+miEFZTKqfCUM6K7xSMQL9OKL/b6hQv+e19PK+JZNE=
-google.golang.org/protobuf v1.20.1 h1:ESRXHgpUBG5D2I5mmsQIyYxB/tQIZfSZ8wLyFDf/N/U=
-google.golang.org/protobuf v1.20.1/go.mod h1:KqelGeouBkcbcuB3HCk4/YH2tmNLk6YSWA5LIWeI/lY=
```
Comment (Marton6) [inline go.sum]: We probably want to commit the new `go.sum` file. I'm not sure why we didn't do that yet

File: secrethub.i [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	time CreatedAt;
+};
+
+extern struct SecretVersion {
+	uuid SecretVersionID;
+	struct Secret Secret;
+    int Version;
+    char* Data;
+    time CreatedAt;
+	char* Status;
```
Comment (Marton6) [inline secrethub.i]: The formatting here should be fixed.

File: README.md [hunk 3]
```diff
@@ -1,73 +1 @@
...
-- https://github.com/vladimirvivien/go-cshared-examples/
-- https://github.com/draffensperger/go-interlang
-- https://golang.org/cmd/cgo/
-
-## Getting Help
-
-Come chat with us on [Discord](https://discord.gg/EQcE87s) or email us at [support@secrethub.io](mailto:support@secrethub.io)
-
-
+# SecretHub XGO
```
Comment (Marton6) [inline README.md:1]: We should add a short description of the project and a section with links to all the language-specific repos (currently only dotnet).

File: go.sum (hunk 1 above)
Comment (edif2008) [reply go.sum]: Do we want to add the `go.sum`? That file is made automatically when building the cshared library.

Comment (Marton6) [reply go.sum]: Yes! See: https://github.com/golang/go/wiki/Modules#should-i-commit-my-gosum-file-as-well-as-my-gomod-file

Comment (edif2008) [reply go.sum]: Added.

File: secrethub.i (hunk 2 above)
Comment (edif2008) [reply secrethub.i]: Fixed.

File: README.md (hunk 3 above)
Comment (edif2008) [reply README.md:1]: Added.

**Comment by mackenbach** [comment]
@Marton6 or @edif2008 could we add a "Roadmap" section to the README? 

### Roadmap
- [ ] Java
- [ ] NodeJS
- [ ] PHP
- [ ] Python
- [ ] Ruby

**Comment by edif2008** [comment]
> @Marton6 or @edif2008 could we add a "Roadmap" section to the README?
> 
> ### Roadmap
> * [ ]  Java
> * [ ]  NodeJS
> * [ ]  PHP
> * [ ]  Python
> * [ ]  Ruby

Added

**Comment by Marton6** [comment]
This should be merged before we release the beta version of the SecretHub .NET client.

**Review by jpcoenen — APPROVED** [review]

**Pull request merged**